monitoring-dashboard/components/org.wso2.micro.integrator.dashboard.web/web-app/src/home/NodeFilter.js

```diff
@@ -62,24 +62,34 @@ function MultipleSelect(props) {
     const [selectedAll, setSelectedAll] = useState(false);
     const dispatch = useDispatch();
 
-    React.useEffect(()=>{
+    useEffect(()=>{
         setSelectedNodeList([])
     },[globalGroupId]);
 
-    React.useEffect(()=>{
+    useEffect(()=>{
         if (nodeList.length !== 0 && selectedNodeList.length == 0) {
             setSelectedAll(true)
             setSelectedNodeList(nodeList);
             dispatch(filterNodes(nodeList))
         }
     },[nodeList]);
 
+    useEffect(()=> {
+        dispatch(filterNodes(selectedNodeList));
+    },[selectedNodeList])
+
     const handleChange = (event) => {
-        setSelectedAll(false);
-        setSelectedNodeList(event.target.value);
+        const selectAllLabel = "select all";
+        const selectAllValueArray = event.target.value.filter(nodeName => nodeName === selectAllLabel)
+        if (selectAllValueArray.length !== 0) {
+           handleSelectAllChange();
+        }else {
```

Comment:
shall we have space in between } and else

Reply:
Fixed it